src/not-so-smart/smart_git.ml

```diff
@@ -383,6 +383,21 @@ struct
         | None -> get_transmission r)
     | [] -> None
 
+  let add_unless lst k v = match List.assoc_opt k lst with
```

Comment:
Are the keys assumed to be lower case? Ideally, we should match case insensitive.